### PR TITLE
P2P Scaffolding

### LINKED ISSUE #171: Initial P2PService Implementation
### Description
Implement the scaffolding code needed for Artemis to interact with the P2PNetwork.

### Acceptance Criteria
* P2PNetwork scaffolding code

### PR DESCRIPTION
**Pull request by jrhea**
## PR description
Put in place the initial scaffolding code for the P2PService.  Uses MockP2PNetwork to generate new block events.
## Fixed Issue(s)

resolves #171

### CODE REVIEW

File: artemis/src/main/java/tech/pegasys/artemis/services/ServiceController.java [hunk 1]
```diff
@@ -25,22 +28,24 @@
       ServiceFactory.getInstance(BeaconChainService.class).getInstance();;
   private static final PowchainService powchainService =
       ServiceFactory.getInstance(PowchainService.class).getInstance();
+  private static final P2PService p2pService =
+      ServiceFactory.getInstance(P2PService.class).getInstance();
   private static final ExecutorService beaconChainExecuterService =
       Executors.newSingleThreadExecutor();
   private static final ExecutorService powchainExecuterService =
       Executors.newSingleThreadExecutor();
+  private static final ExecutorService p2pExecuterService = Executors.newSingleThreadExecutor();
```
Comment (schroedingerscode) [inline artemis/src/main/java/tech/pegasys/artemis/services/ServiceController.java:37]: I guess it's just under the column limit, so spotless didn't care, but let's try this for consistency.
```suggestion
  private static final ExecutorService p2pExecuterService =
      Executors.newSingleThreadExecutor();
```

File: artemis/src/main/java/tech/pegasys/artemis/services/ServiceController.java [hunk 2]
```diff
@@ -59,5 +65,6 @@ public static void stopAll(CommandLineArguments cliArgs) {
     beaconChainService.stop();
     powchainExecuterService.shutdown();
     powchainService.stop();
+    p2pService.stop();
```
Comment (schroedingerscode) [inline artemis/src/main/java/tech/pegasys/artemis/services/ServiceController.java:69]: Should p2pExecutorService.shutdown() also be called here?

File: artemis/src/main/java/tech/pegasys/artemis/services/p2p/P2PService.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import tech.pegasys.artemis.services.ServiceInterface;
+
+public class P2PService implements ServiceInterface {
+
+  private EventBus eventBus;
+  private static final Logger LOG = LogManager.getLogger();
+  private P2PNetwork p2pNetwork;
+
+  public P2PService() {
+    this.eventBus = new AsyncEventBus(Executors.newCachedThreadPool());
```
Comment (schroedingerscode) [inline artemis/src/main/java/tech/pegasys/artemis/services/p2p/P2PService.java]: This seems to conflict with Line 37. We should either create an event bus internally or pass the dependency in. Both is odd.

File: artemis/src/main/java/tech/pegasys/artemis/services/p2p/P2PService.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  @Override
+  public void init(EventBus eventBus) {
```
Comment (schroedingerscode) [inline artemis/src/main/java/tech/pegasys/artemis/services/p2p/P2PService.java:33]: This seems to conflict with Line 33. We should either create an event bus internally or pass the dependency in. Both is odd.

**Review by schroedingerscode — CHANGES_REQUESTED** [review]
- Some styling nitpicks.
 - I believe cleaning up the p2p executor service was missed.
 - General concerns about the way EventBuses are passed through to the services.

File: artemis/src/main/java/tech/pegasys/artemis/services/ServiceController.java (hunk 2 above)
Comment (jrhea) [reply artemis/src/main/java/tech/pegasys/artemis/services/ServiceController.java:69]: great catch.  thanks!

File: artemis/src/main/java/tech/pegasys/artemis/services/p2p/P2PService.java (hunk 3 above)
Comment (jrhea) [reply artemis/src/main/java/tech/pegasys/artemis/services/p2p/P2PService.java]: you know as well as i do why this happened.  good catch though!!!!

File: artemis/src/main/java/tech/pegasys/artemis/services/p2p/P2PService.java (hunk 4 above)
Comment (jrhea) [reply artemis/src/main/java/tech/pegasys/artemis/services/p2p/P2PService.java:33]: i need to lay off the spec.  thanks

**Review by schroedingerscode — APPROVED** [review]
LGTM

**Pull request merged**